src/frontend/dash/circlePacker.ts

```diff
@@ -30,101 +38,149 @@ export class MemberCircle {
-    placeCircleFromPoint(circle, 0, 0, angle, d)
-
-    // move inwards until touching
-    let collided = null
-    while (d > 0 && !collided) {
-        collided = getCollided(circle, MARGIN);
-        if (collided || d <= 0) { break }
-        d -= STEP_DISTANCE;
-        placeCircleFromPoint(circle, 0, 0, angle, d)
-        // redrawCircles(placedCircles.concat([circle])) // REDRAWWWWW = = = == = = = == = == = = = == 
-        // await sleep(10) /// REDRAWWWWW
+    // TODO: remove from placeCircle and onto placeCircles
+    
+    if(!placedCircles.length) {
+        circle.x = circle.y = 0;
+        return;
     }
 
-    // rotate around collided circle
-    let STEP_ANGLE = (STEP_DISTANCE / collided.r) * Math.PI * 2
-    STEP_ANGLE = STEP_ANGLE * (Math.random() < 0.5 ? 1 : -1) // randomize direction | -1/1 credit https://stackoverflow.com/questions/29109456/how-to-get-random-1-or-1-in-canvas-javascript
-    let A = getAngleFromAToB(collided, circle)
-    let a;
-    for (a = A; a < A + Math.PI * 2 && a > A - Math.PI * 2; a += STEP_ANGLE) {
-        placeCircleFromPoint(circle, collided.x, collided.y, a, collided.r + circle.r + MARGIN + .1)
-        if (getCollided(circle, MARGIN)) { console.log(circle); break }
-        // redrawCircles(placedCircles.concat([circle])) // REDRAWWWWW = = = == = = = == = == = = = == 
-        // await sleep(10) /// REDRAWWWWW
-    }
-    // rotate out of collided circle
-    for (A = a; Math.abs(A - a) < Math.abs(STEP_ANGLE); a -= (STEP_ANGLE / 5)) {
-        console.log('doing')
-        placeCircleFromPoint(circle, collided.x, collided.y, a, collided.r + circle.r + MARGIN + .1)
-        if (!getCollided(circle, MARGIN)) { break }
-
-        // redrawCircles(placedCircles.concat([circle])) // REDRAWWWWW = = = == = = = == = == = = = == 
-        // await sleep(10) /// REDRAWWWWW
+    if(placedCircles.length == 1) {
+        // circle.touching[circle.touching.length] = placedCircles[0].r, but stupider
+        //^^^what's the deal with the length being an index here?^^^ -- thing[thing.length] is append to the last index
+
+        const distance = Math.pow(MARGIN + placedCircles[0].r + circle.r, 2);
+
+        const rand = Math.random();
+
+        circle.x = Math.round(Math.random()) ? Math.sqrt(rand * distance) : -Math.sqrt(rand * distance);
+        circle.y = Math.round(Math.random()) ? Math.sqrt((1 - rand) * distance) : -Math.sqrt((1 - rand) * distance);
+        return;
     }
+    
+    for(let index1 = 0; index1 != placedCircles.length; index1++) {
+        const circle1 = placedCircles[index1];
+        for(let index2 = index1+1; index2 != placedCircles.length; index2++) {
+            const circle2 = placedCircles[index2];
+            const distanceFrom = circle1.r + circle2.r + MARGIN; 
+
+            const radius1 = circle1.r + circle.r + MARGIN;
+            const radius2 = circle2.r + circle.r + MARGIN;
+
+            // a : distance from fulcrum
```

Comment:
What do you mean by fulcrum in this context?

Reply:
Distance to the point on a line that intersects the centers of the two circles that's also perpendicular to the radii. Basically where both circles will have an equal position.

Reply:
So... the point between two circles?

Reply:
Maybe a more helpful name would be better

Reply:
<img width="326" alt="image" src="https://user-images.githubusercontent.com/78449263/200736631-868ea402-e644-4552-8fed-64b09a2922e4.png">

Reply:
So the "fulcrum" is the purple line?

Reply:
ye